File: prob-2.java
//TC:O(LOG(N-K))+O(K)
//SC:O(1)
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) 
    {
       int low = 0;
       int high = arr.length-k;
       while(low<high){
          int mid = low+(high-low)/2;
          int lb = Math.abs(x-arr[mid]);
          int hb = Math.abs(arr[mid+k]-x);
          if(lb <= hb)
             high=mid;
          else
             low=mid+1;
       }
        
       
        List<Integer> l = new ArrayList<>();
       for(int i=low;i<low+k;i++)
           l.add(arr[i]);
        
        
       return l; 
    }
}

File: prob2.py
# Time Complexity :O(log(n-k))
# Space Complexity :O(1)
# Leet Code: Yes

class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        low = 0
        high = len(arr) - k
        res = []

        while low < high:
            mid = low + (high-low)//2
            s = x - arr[mid]
            e = arr[mid+k] - x

            if s > e:
                low = mid+1
            else:
                high = mid
        # print(low)
        
        for i in range(k):
            res.append(arr[low+i])

        return res